### DISCUSSION TITLE
Preview deploy of frontend and backend

### DISCUSSION THREAD

**Discussion by pawelshopstory** [vercel/community#1727] [Help]
### Summary

Hello everyone!

I've Nx monorepo which consists of our frontend demo app where we test our product, but also backend for that product. Both of them are Next apps. If I submit a pull request which modifies both of apps, I will have deploy preview for each of them, but the problem is that preview frontend app won't make calls to preview of our backend. Is there any best practise to achieve that? My current solution is calling the preview URL (it's predictable [appname]-git-[branch-name].vercel.app) and if that fails, that means there is no deploy preview of my backend I fallback to production one.

Cheers!

**Comment by amyegan** [comment]
Hi @pawelshopstory! Your current solution using the [git branch generated URL](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) is what I would suggest for this situation. You can use the `VERCEL_GIT_COMMIT_REF` [system environment variable](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) to get the branch name if you aren't already doing that. The rest of the URL, as you said, is made up of predicable [components](https://vercel.com/docs/concepts/deployments/generated-urls#url-components) in the format of `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`

**Reply by pawelshopstory** [reply]
Hey @amyegan !

Thank you for your response. I am happy that my solution is aligned with yours.

Do you have any plans on introducing such behaviour as a built in option? My first guess is that maybe there could be an option to tell that my frontend project depends on backend project and when backend has preview deploy from the same branch, then there should be additional environment variables set to have an easy access to that project preview URL.
